velox/functions/prestosql/tests/CastBaseTest.h

```diff
@@ -67,25 +71,37 @@ class CastBaseTest : public FunctionBaseTest {
     assertEqualVectors(expected, result);
   }
 
+  core::TypedExprPtr buildCastExprDictEncoding(
+      const TypePtr& fromType,
+      const TypePtr& toType,
+      bool tryCast = false) {
+    core::TypedExprPtr inputField =
+        std::make_shared<const core::FieldAccessTypedExpr>(fromType, "c0");
+    core::TypedExprPtr callExpr = std::make_shared<const core::CallTypedExpr>(
+        fromType,
+        std::vector<core::TypedExprPtr>{inputField},
+        "testing_dictionary");
+    return std::make_shared<const core::CastTypedExpr>(
+        toType, std::vector<core::TypedExprPtr>{callExpr}, tryCast);
+  }
+
+  void verifyCastDictEncoding(
```

Comment:
This method is a bit cryptic. It is confusing to pass 'expected' values, but then modify it by applying a very particular type of dictionary encoding. It might be clearer to just move this logic to the caller.